examples/functions/golang/serving-mmlogic-simple/main.go

```diff
@@ -0,0 +1,135 @@
+/*
+This application handles all the startup and connection scaffolding for
+running a gRPC server serving the APIService as defined in
+${OM_ROOT}/internal/pb/function.pb.go
+
+NOTE: This code will likely all get moved refactoring is done and the code
+for spinning up a gRPC API is DRYed out.
+
+All the actual important bits are in the API Server source code: apisrv/
+
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"os/signal"
+
+	"github.com/GoogleCloudPlatform/open-match/config"
+	"github.com/GoogleCloudPlatform/open-match/examples/functions/golang/serving-mmlogic-simple/apisrv"
+	"github.com/GoogleCloudPlatform/open-match/internal/metrics"
+	api "github.com/GoogleCloudPlatform/open-match/internal/pb"
+	redisHelpers "github.com/GoogleCloudPlatform/open-match/internal/statestorage/redis"
+	"go.opencensus.io/plugin/ocgrpc"
+	"google.golang.org/grpc"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+var (
+	// Logrus structured logging setup
+	fnLogFields = log.Fields{
+		"app":       "openmatch",
+		"component": "function_service",
+	}
+	fnLog = log.WithFields(fnLogFields)
+
+	// Viper config management setup
+	cfg = viper.New()
+	err = errors.New("")
+)
+
+func init() {
```

Comment:
Fold this into Run().

A later pull request will canonicalize server initialization for 2 reasons, simplified e2e server bootstraping and adding gRPC secured mode.

For example: https://github.com/jeremyje/open-match/blob/alignment/internal/app/backendapi/backendapi.go.

Reply:
thanks for the example, sorted